src/datasets/arrow_dataset.py

```diff
@@ -1264,7 +1264,12 @@ def format_cache_file_name(cache_file_name, rank):
                 logger.info("Process #{} will write at {}".format(rank, cache_file_name))
                 return cache_file_name
 
+            prev_env = dict(os.environ)
+            if prev_env.get("TOKENIZERS_PARALLELISM", "false") != "false":
```

Comment:
Are you sure that checking `"false"` is enough?
In `transformers` we check various way to write env variable values: https://github.com/huggingface/transformers/blob/master/src/transformers/file_utils.py#L52-L55

Reply:
Well `false` is different from anything in `("1", "ON", "YES", "AUTO")` :)
But for consistency I'll use the same thing instead of checking for `false`

Reply:
Not sure about "AUTO" though

Reply:
maybe you can convert to `lower` before checking for `"false"` and check for `0` as well?

Reply:
I checked `tokenizers` source code to check for every way of saying "no" to parallelism.
It should be good now

Reply:
I was going to send you our list from `tokenizers`, you were faster!